wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift

```diff
@@ -337,20 +278,79 @@ extension ConversationCreationController: AddParticipantsConversationCreationDel
             values.participants = users
 
         case .create:
-            var allParticipants = values.participants
-            allParticipants.insert(selfUser)
+            guard let userSession = ZMUserSession.shared() else { return }
+            let service = ConversationService(context: userSession.viewContext)
 
-            delegate?.conversationCreationController(
-                self,
-                didSelectName: values.name,
-                participants: values.participants,
+            let users = values.participants
+                .union([selfUser])
+                .materialize(in: userSession.viewContext)
+
+            service.createGroupConversation(
+                name: values.name,
+                users: Set(users),
                 allowGuests: values.allowGuests,
                 allowServices: values.allowServices,
                 enableReceipts: values.enableReceipts,
-                encryptionProtocol: values.encryptionProtocol
-            )
+                messageProtocol: values.encryptionProtocol == .proteus ? .proteus : .mls
+            ) { [weak self] in
+                guard let self = self else { return }
+
+                switch $0 {
+                case .success(let conversation):
+                    showNonFederatingDomainsAlert(domains: [
+                        "foo.com",
+                        "bar.com",
+                        "goo.com"
+                    ])
+                    return
+//                    delegate?.conversationCreationController(
+//                        self,
+//                        didCreateConversation: conversation
+//                    )
```

Comment:
Do we still need this? Or we can get rid of?

Reply:
The PR isn't finished yet.